### ISSUE TITLE
Low validation mIoU despite high validation accuracy

### ISSUE THREAD

**Issue by mackenzie-warren** [lijx10/SO-Net#37]
I am getting the same error as mention in a previous issue: https://github.com/lijx10/SO-Net/issues/34

The validation mIoU is only reaching 0.16, although the validation accuracy is about 90%.  The was run with the default parameters in the code and using the prepared version of ShapeNet found here: https://drive.google.com/open?id=184MbflF_RbDX9MyML3hid7OxsYJ8oQQ7

I've attached a plot that shows the train and validation losses and accuracy, and the validation mIoU, at every epoch during training. 

<img width="414" alt="Screen Shot 2021-02-22 at 9 00 02 AM" src="https://user-images.githubusercontent.com/7797698/108730333-6ed40700-74f9-11eb-9d01-30e6233b2ea0.png">

**Comment by mackenzie-warren** [comment]
Fix suggested in pull request #38: in calculation of IoU, the tensors need to be converted to int from boolean before adding.  E.g. lines 155 in `losses.py` should be:

`intersection = (gt.int() + predict.int()) == 2
union = (gt.int() + predict.int()) >= 1`
